### ISSUE TITLE
BMK module compilation skips modules

### ISSUE THREAD

**Issue by GWRon** [bmx-ng/bmk#39]
```
C:\Programmieren\BlitzMaxNG\bin>bmk makemods -d -a -g x64 brl
[  0%] Processing:blitz.bmx
[  0%] Processing:debugger_mt.stdio.bmx
[  0%] Processing:sample.bmx
[  0%] Processing:errors.bmx
[  0%] Processing:iterator.bmx
[  1%] Processing:d3d7max2d.bmx
[  1%] Processing:data.bmx
[  1%] Processing:font.bmx
[  1%] Processing:hook.bmx
[  1%] Processing:keycodes.bmx
[  1%] Processing:intmap.bmx
[  2%] Processing:objectmap.bmx
[  2%] Processing:ptrmap.bmx
[  2%] Processing:stringmap.bmx
[  2%] Processing:math.bmx
[  2%] Processing:pixel.bmx
[  2%] Processing:random.bmx
[  3%] Processing:common.bmx
[  3%] Processing:driver.bmx
[  3%] Processing:threads.bmx
[  3%] Processing:d3d.bmx
[  3%] Processing:freeaudio.bmx
[  6%] Processing:libjpeg.bmx
[  6%] Processing:lua.bmx
[  6%] Processing:oggvorbis.bmx
bcc[ng] Release Version 0.93
[  6%] Processing:openal.bmx
[  7%] Processing:stbimage.bmx
[  7%] Processing:com.bmx
[  7%] Processing:commdlg.bmx
[  7%] Processing:gdi32.bmx
[  8%] Processing:kernel32.bmx
[  8%] Processing:shell32.bmx
[  8%] Processing:user32.bmx
[  8%] Processing:winmm.bmx
[  8%] Processing:zlib.bmx
[  8%] Processing:appstub.bmx
[  9%] Processing:collection.bmx
[  9%] Processing:map.bmx
[  9%] Processing:stringbuilder.bmx
[  9%] Processing:commctrl.bmx
[ 10%] Processing:richedit.bmx
[ 10%] Processing:linkedlist.bmx
[ 10%] Processing:socket.bmx
[ 10%] Processing:stream.bmx
[ 10%] Processing:enet.bmx
[ 10%] Processing:source.bmx
[ 11%] Processing:libpng.bmx
[ 11%] Processing:queue.bmx
[ 11%] Processing:win32.bmx
[ 11%] Processing:freetype.bmx
[ 11%] Processing:audiosample.bmx
[ 12%] Processing:bank.bmx
[ 12%] Processing:list.bmx
[ 12%] Processing:endianstream.bmx
[ 12%] Processing:event.bmx
[ 12%] Processing:pixmap.bmx
[ 12%] Processing:ramstream.bmx
[ 13%] Processing:reflection.bmx
[ 13%] Processing:socketstream.bmx
[ 13%] Processing:textstream.bmx
[ 13%] Processing:threadpool.bmx
[ 13%] Processing:d3d7.bmx
[ 13%] Processing:d3d9x.bmx
[ 14%] Processing:dd.bmx
[ 14%] Processing:dsound.bmx
[ 14%] Processing:collections.bmx
[ 14%] Processing:audio.bmx
[ 14%] Processing:bankstream.bmx
[ 14%] Processing:freetypefont.bmx
[ 15%] Processing:httpstream.bmx
[ 15%] Processing:jpgloader.bmx
[ 15%] Processing:maxlua.bmx
[ 15%] Processing:oggloader.bmx
[ 15%] Processing:pngloader.bmx
[ 15%] Processing:standardio.bmx
[ 16%] Processing:stbimageloader.bmx
[ 16%] Processing:timer.bmx
[ 16%] Processing:wavloader.bmx
[ 16%] Processing:bmploader.bmx
[ 16%] Processing:filesystem.bmx
[ 17%] Processing:openalaudio.bmx
[ 17%] Processing:tgaloader.bmx
[ 17%] Processing:d3d9.bmx
[ 17%] Processing:directx.bmx
[ 17%] Processing:basic.bmx
[ 17%] ProcessCompile Error: Can't find interface for module 'brl.random'
[C:/Programmieren/BlitzMaxNG/mod/brl.mod/basic.mod/basic.bmx;18;0]
ing:maxutil.bmx
[ 18%] Processing:sBuild Error: failed to compile (-1) C:/Programmieren/BlitzMaxNG/mod/brl.mod/basic.mod/basic.bmxystem.
bmx

C:\Programmieren\BlitzMaxNG\bin>
```

It also seems to eat some text output  ("ing:maxutil.bmx"" "bmxystem.bmx"). I somehow think the threading is guilty.

**Comment by woollybah** [comment]
I still haven't managed to replicate this.

**Comment by GWRon** [comment]
Maybe it is a good solution to have BMK create a bmk.debug.log file when compiled in debug mode. So to narrow down things?
I suggested this already some time ago but what about tackling it now?

You know best, what it should output and what not (thinking of "files to compile", "steps" ...). It might even help optimizing stuff (linear vs non-linear).

I would suggest to have multiple log files (so you are not left with an empty file on crash):
- bmk.debug.log:
  - contains whole bmk stuff like "arguments: xxx", "files to compile: x..."
- bmk.debug.raw.log (optional):
  - contains what gets output (in that order)

Why? If you wrote everything in one single file, threads might interfer. Writing a new file for each thread is also not a good idea as you start a bunch of threads during bmk execution.

**Comment by GWRon** [comment]
Experienced it again - on a Windows 7 desktop machine, on a windows 10 laptop and a windows XP VM.

```
[ 51%] Processing:spektroNG.bmx
[ 79%] Compiling:string_comp.c
[ 80%] Compiling:glue.c
[ 80%] Compiling:common.bmx.debug.win32.x86.c
[ 80%] Compiling:common.bmx.debug.win32.x86.c
[ 81%] Compiling:wxdataobject.bmx.debug.win32.x86.c
[ 81%] Compiling:wxdatetime.bmx.debug.win32.x86.c
[ 82%] Compiling:common.bmx.debug.win32.x86.c
In file included from E:/BlitzMaxNG.win32/mod/wx.mod/wxdataobject.mod/.bmx/common.bmx.debug.win32.x86.c:1:0:
E:/BlitzMaxNG.win32/mod/wx.mod/wxdataobject.mod/.bmx/common.bmx.debug.win32.x86.h:5:10: fatal error: wx.mod/wx.mod/.bmx/wx.bmx.debug.win32.x86.h: No such file or directory
 #include <wx.mod/wx.mod/.bmx/wx.bmx.debug.win32.x86.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
Build Error: failed to compile (1) E:/BlitzMaxNG.win32/mod/wx.mod/wxdataobject.mod/.bmx/common.bmx.debug.win32.x86.c
Process complete
```
This was compiled without "quick" etc. 
Needed to manually trigger a "bmk makemods -a -d wx" to get them compiled.
Before it blamed to not find brl.mod/basic...-stuff. Had then compiled them via "bmk makemods -a -d brl".

**Comment by woollybah** [comment]
I've no idea why it doesn't work for you.
If the generated .h is missing, it should force a compile of the .bmx file. This should occur before 
wxdataobject because wx is a dependency of it.

It's almost as if FileTime() is incorrectly reporting large values for you. It should return 0 for non existent files...

**Comment by woollybah** [comment]
Assumed fixed in 693999dd340c7feaf17d7dd7b9fec647b1782e03.
